server/backend/database/memory/database.go

```diff
@@ -647,24 +653,36 @@ func (d *DB) FindDocInfoByKeyAndOwner(
 
 	now := gotime.Now()
 	var docInfo *database.DocInfo
+	newDocInfo := &database.DocInfo{
+		ID:         newID(),
+		ProjectID:  projectID,
+		Key:        key,
+		Owner:      clientID,
+		ServerSeq:  0,
+		CreatedAt:  now,
+		AccessedAt: now,
+	}
+
 	if raw == nil {
-		docInfo = &database.DocInfo{
-			ID:         newID(),
-			ProjectID:  projectID,
-			Key:        key,
-			Owner:      clientID,
-			ServerSeq:  0,
-			CreatedAt:  now,
-			AccessedAt: now,
-		}
+		docInfo = newDocInfo
 		if err := txn.Insert(tblDocuments, docInfo); err != nil {
 			return nil, fmt.Errorf("create document: %w", err)
 		}
-		txn.Commit()
 	} else {
 		docInfo = raw.(*database.DocInfo)
+		if !docInfo.RemovedAt.IsZero() {
+			if err := txn.Delete(tblDocuments, docInfo); err != nil {
```

Comment:
Sorry for my additional comment, I'm a bit confused about this code.

If `createDocIfNotExist` is `false` and raw is not nil (there is `docInfo` that has been marked as removed), this code will delete previous `docInfo` and create new `docInfo`.

Why is it deleting and creating `docInfo` even `createDocIfNotExist` is false?

Also, it may be dangerous to delete only `docInfo`, since `doc_id` in `docInfo` is referenced in other collections. Reference about deleting `document` is mentioned below. 

<p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; color: rgb(230, 237, 243); font-family: -apple-system, &quot;system-ui&quot;, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><strong style="box-sizing: border-box; font-weight: var(--base-text-weight-semibold, 600);">Data to be deleted</strong>:<br style="box-sizing: border-box;">To completely delete a document, all data related to the document in the server and DB should be eliminated.</p>

struct | DB column | DB related key
-- | -- | --
DocInfo | documents |  
ClientDocInfo (in ClientInfo) | clients | documents.${docID}.${*}
ChangeInfo | changes | doc_id
SnapshotInfo | snapshots | doc_id
SyncedSeqInfo | syncedseqs | doc_id